client/web/src/site-admin/SiteAdminExternalServicesArea.tsx

```diff
@@ -0,0 +1,112 @@
+    'ExternalServicePage'
+)
+
+const AddExternalServicesPage = lazyComponent(
+    () => import('../components/externalServices/AddExternalServicesPage'),
+    'AddExternalServicesPage'
+)
+
+interface Props
+    extends RouteComponentProps<{}>,
+        ThemeProps,
+        TelemetryProps,
+        PlatformContextProps,
+        SettingsCascadeProps {
+    authenticatedUser: AuthenticatedUser
+}
+
+export const SiteAdminExternalServicesArea: React.FunctionComponent<React.PropsWithChildren<Props>> = ({
+    match,
+    ...outerProps
+}) => {
+    const { data, error, loading } = useQuery<SiteExternalServiceConfigResult, SiteExternalServiceConfigVariables>(
+        SITE_EXTERNAL_SERVICE_CONFIG,
+        {}
+    )
+
+    if (error && !loading) {
+        return <ErrorAlert error={error} />
+    }
+
+    if (loading && !error) {
+        return <LoadingSpinner />
+    }
+
+    if (!data) {
+        return null
+    }
+
+    return (
+        <Switch>
+            <Route
+                path={match.url}
+                render={props => (
+                    <ExternalServicesPage
+                        {...outerProps}
+                        {...props}
+                        routingPrefix="/site-admin"
+                        afterDeleteRoute="/site-admin/repositories?repositoriesUpdated"
+                        externalServicesFromFile={data?.site?.externalServicesFromFile}
+                        allowEditExternalServicesWithFile={data?.site?.allowEditExternalServicesWithFile}
+                    />
+                )}
+                exact={true}
+            />
+            <Route path={match.url + '/add'} render={() => <Redirect to="new" />} exact={true} />
+            <Route
+                path={`${match.url}/new`}
+                render={props => (
+                    <AddExternalServicesPage
+                        {...outerProps}
+                        {...props}
+                        routingPrefix="/site-admin"
+                        afterCreateRoute="/site-admin/repositories?repositoriesUpdated"
+                        codeHostExternalServices={codeHostExternalServices}
+                        nonCodeHostExternalServices={nonCodeHostExternalServices}
+                    />
+                )}
+                exact={true}
+            />
+            <Route
+                path={`${match.url}/:id`}
+                render={({ match, ...props }: RouteComponentProps<{ id: Scalars['ID'] }>) => (
+                    <ExternalServicePage
+                        {...outerProps}
+                        {...props}
+                        externalServiceID={match.params.id}
+                        afterUpdateRoute="/site-admin/repositories?repositoriesUpdated"
+                        externalServicesFromFile={data?.site?.externalServicesFromFile}
+                        allowEditExternalServicesWithFile={data?.site?.allowEditExternalServicesWithFile}
+                    />
+                )}
```

Comment:
I imagine Buildkite will still complain here, you'll need to do something like

```suggestion
                path={`${match.url}/:id`}
                render={({ match, ...props }: RouteComponentProps<{ id?: Scalars['ID'] }>) => (
                    <ExternalServicePage
                        {...outerProps}
                        {...props}
                        externalServiceID={match.params.id as Scalars['ID']}
                        afterUpdateRoute="/site-admin/repositories?repositoriesUpdated"
                        externalServicesFromFile={data?.site?.externalServicesFromFile}
                        allowEditExternalServicesWithFile={data?.site?.allowEditExternalServicesWithFile}
                    />
                )}
```

since RouteComponentProps are always optional

Reply:
Or do some other form of "not undefined" assertion

Reply:
umm I don't thin they should be optional, where's that from? I've always made them non-optional.

Reply:
Hmmmm my IDE complains about this one, but not about other ones. And if I run `yarn build-ts` it also complains about it. Not sure what's going on on my side then, apologies

Reply:
Oh, I did not pull the latest changes 🤦‍♂️ where the `path=` was changed.